src/commands/ipmi.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+#==============================================================================
+# Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
+#
+# This file/package is part of Alces Metalware.
+#
+# Alces Metalware is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation, either version 3 of
+# the License, or (at your option) any later version.
+#
+# Alces Metalware is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this package.  If not, see <http://www.gnu.org/licenses/>.
+#
+# For more information on the Alces Metalware, please visit:
+# https://github.com/alces-software/metalware
+#==============================================================================
+
+require 'command_helpers/base_command'
+require 'command_helpers/node_identifier'
+require 'config'
+
+module Metalware
+  module Commands
+    class Ipmi < CommandHelpers::BaseCommand
+      private
+
+      prepend CommandHelpers::NodeIdentifier
+
+      def setup
+        @options = options
+        @args = args
+      end
+
+      def run
+        ipmi(command)
+      end
+
+      def ipmi(cmd)
+        system(cmd.to_s)
+      end
+
+      def command
+        "ipmitool #{render_hosts} #{render_credentials} #{render_command}"
+      end
+
+      def render_command
+        @options.command
+      end
+
+      def render_hosts
+        if @options.group
+          "-g #{@args[0]}"
+        else
+          "-H #{render_hostname}"
+        end
+      end
+
+      def render_credentials
+        if @options.group
+          username = alces.domain.config.networks.bmc.bmcuser
+          password = alces.domain.config.networks.bmc.bmcpassword
```

Comment:
Is this strictly correct? If we're running the command for a node we will use the node's BMC username and password (sounds reasonable), but if we're running it for a group we will use the whole domain's username and password - but couldn't these be set specifically for some particular group distinct from the whole domain?

Reply:
Yep, you are correct! I'll change this to look for a groups config instead, thanks :)

Reply:
Amazing, thanks. I can't be certain but I'd say it's quite an edge case - @ColonelPanicks probably has a better idea if that's likely to occur or not. Typically all nodes in a cluster will have the same BMC user/password though.